### PR TITLE
fb: cfb: add API to set font kerning

### PR DESCRIPTION
**Pull request by 5frank**
Font kerning was used but it was not possible to change it. Also, font_idx was set to zero twice in init.

Signed-off-by: Simon Frank <simon.frank@lohmega.com>

### CODE REVIEW

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please change commit title to `fb: cfb: add API to set font kerning `, thanks

**Review by jfischer-no — DISMISSED** [review]

File: subsys/fb/cfb.c [hunk 1]
```diff
@@ -343,7 +351,6 @@ int cfb_framebuffer_init(const struct device *dev)
 	fb->pixel_format = cfg.current_pixel_format;
 	fb->screen_info = cfg.screen_info;
 	fb->buf = NULL;
-	fb->font_idx = 0U;
```
Comment (soburi) [inline subsys/fb/cfb.c:346]: Is it need to remove?

File: include/zephyr/display/cfb.h [hunk 2]
```diff
@@ -153,6 +153,16 @@ int cfb_get_display_parameter(const struct device *dev,
...
 
+/**
+ * @brief Set font kerning (spacing between individual letters).
+ *
+ * @param dev Pointer to device structure for driver instance
+ * @param kerning Font kerning
+ *
+ * @return 0 on success, negative value otherwise
+ */
+int cfb_set_kerning(const struct device *dev, int8_t kerning);
```
Comment (soburi) [inline include/zephyr/display/cfb.h:164]: Could you please add command to shell that invoke the API?

File: subsys/fb/cfb.c [hunk 3]
```diff
@@ -316,6 +316,14 @@ int cfb_get_font_size(const struct device *dev, uint8_t idx, uint8_t *width,
...
 }
 
+int cfb_set_kerning(const struct device *dev, int8_t kerning)
+{
+	char_fb.kerning = kerning;
+
+	return 0;
+}
+
+
```
Comment (soburi) [inline subsys/fb/cfb.c:326]: Blank line is usually 1-line

**Review by soburi — CHANGES_REQUESTED** [review]

File: subsys/fb/cfb.c (hunk 1 above)
Comment (5frank) [reply subsys/fb/cfb.c:346]: yes. same line twice. please see commit message.

Comment (soburi) [reply subsys/fb/cfb.c:346]: Sorry, I missed the comments.
It's no problem.

File: include/zephyr/display/cfb.h (hunk 2 above)
Comment (5frank) [reply include/zephyr/display/cfb.h:164]: added command `set_kerning` to cfb_shell.c

File: subsys/fb/cfb_shell.c [hunk 4]
```diff
@@ -235,6 +235,33 @@ static int cmd_set_font(const struct shell *shell, size_t argc, char *argv[])
 	return err;
 }
 
+static int cmd_set_kerning(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+	char *ep = NULL;
+	long int kerning;
```
Comment (soburi) [inline subsys/fb/cfb_shell.c:242]: Please Fix CI check.

File: subsys/fb/cfb_shell.c [hunk 5]
```diff
@@ -235,6 +235,33 @@ static int cmd_set_font(const struct shell *shell, size_t argc, char *argv[])
 	return err;
 }
 
+static int cmd_set_kerning(const struct shell *shell, size_t argc, char *argv[])
```
Comment (soburi) [inline subsys/fb/cfb_shell.c:238]: Could you please separate commits about `cfb_shell.c` modification?

**Review by soburi — CHANGES_REQUESTED** [review]

**Review by soburi — APPROVED** [review]

**Comment by soburi** [comment]
@5frank 
LGTM. Thank you for contribution!

@jfischer-no 
Please take a look.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**